### ISSUE TITLE
"Error from OpenKeyChain: null" when inserting new password

### ISSUE THREAD

**Issue by epinna** [zeapo/Android-Password-Store#245]
Hello,

When I try to save a new password it displays the "Error from OpenKeyChain: null" error and the operation fails.

To reproduce:
1. Click the "+" button 
2. Insert random name and password
3. Click ✓to save
4. Error "Error from OpenKeyChain: null" is visualized
5. The fields are blanked and the password is not saved

It's "Password Store (beta)" version 1.2.0.47 and OpenKeyChain version 4.1 both installed from the playstore. The other interactions with OpenKeyChain (e.g. the virtualisation of the passwords in the storage) work correctly.

**Comment by zeapo** [comment]
Hi,

Is it the first time you add a new password?

**Comment by epinna** [comment]
I have a bunch of passwords added from other devices and pulled in the Android app - but is the first time I add new passwords from the app itself indeed.

**Comment by avanc** [comment]
I have the same problem on my device with the following software:
Android: 7.0 (Securityupdates from 5th November)
Android PasswordStore: 1.2.0.25
OpenKeychain: 4.1 (41010)

It only happens while adding new passwords (thus encryption). Decryption works without any problems.

First, I thought I did something wrong while importing the public key. But encryption with OpenKeychain works also.

I created a key pair by specifying an name, but without email address.

One thing I found is that in settings under "Select OpenPGP-Key-ID" no key is selected. If click on this, OpenKeychain is shown. But I cannot select my key "PasswordStore" and the Okay-Button is grey.

Besides, what is the difference between "Select key" and "Select OpenPGP-Key-ID"?

**Comment by cyrus-and** [comment]
Same here, and here's the logcat:

```
12-01 20:51:19.892 21610 21620 E JavaBinder: *** Uncaught remote exception!  (Exceptions are not yet supported across processes.)
12-01 20:51:19.892 21610 21620 E JavaBinder: java.lang.AssertionError: cannot clearsign in non-ascii armored text, this is a bug!
12-01 20:51:19.892 21610 21620 E JavaBinder:    at org.sufficientlysecure.keychain.pgp.PgpSignEncryptOperation.executeInternal(PgpSignEncryptOperation.java:573)
12-01 20:51:19.892 21610 21620 E JavaBinder:    at org.sufficientlysecure.keychain.pgp.PgpSignEncryptOperation.execute(PgpSignEncryptOperation.java:165)
12-01 20:51:19.892 21610 21620 E JavaBinder:    at org.sufficientlysecure.keychain.remote.OpenPgpService.encryptAndSignImpl(OpenPgpService.java:412)
12-01 20:51:19.892 21610 21620 E JavaBinder:    at org.sufficientlysecure.keychain.remote.OpenPgpService.executeInternalWithStreams(OpenPgpService.java:947)
12-01 20:51:19.892 21610 21620 E JavaBinder:    at org.sufficientlysecure.keychain.remote.OpenPgpService.executeInternal(OpenPgpService.java:890)
12-01 20:51:19.892 21610 21620 E JavaBinder:    at org.sufficientlysecure.keychain.remote.OpenPgpService2$1.execute(OpenPgpService2.java:62)
12-01 20:51:19.892 21610 21620 E JavaBinder:    at org.openintents.openpgp.IOpenPgpService2$Stub.onTransact(IOpenPgpService2.java:80)
12-01 20:51:19.892 21610 21620 E JavaBinder:    at android.os.Binder.execTransact(Binder.java:453)
12-01 20:51:19.893 21610 21620 W System.err: java.lang.AssertionError: cannot clearsign in non-ascii armored text, this is a bug!
12-01 20:51:19.893 21610 21620 W System.err:    at org.sufficientlysecure.keychain.pgp.PgpSignEncryptOperation.executeInternal(PgpSignEncryptOperation.java:573)
12-01 20:51:19.893 21610 21620 W System.err:    at org.sufficientlysecure.keychain.pgp.PgpSignEncryptOperation.execute(PgpSignEncryptOperation.java:165)
12-01 20:51:19.893 21610 21620 W System.err:    at org.sufficientlysecure.keychain.remote.OpenPgpService.encryptAndSignImpl(OpenPgpService.java:412)
12-01 20:51:19.893 21610 21620 W System.err:    at org.sufficientlysecure.keychain.remote.OpenPgpService.executeInternalWithStreams(OpenPgpService.java:947)
12-01 20:51:19.893 21610 21620 W System.err:    at org.sufficientlysecure.keychain.remote.OpenPgpService.executeInternal(OpenPgpService.java:890)
12-01 20:51:19.893 21610 21620 W System.err:    at org.sufficientlysecure.keychain.remote.OpenPgpService2$1.execute(OpenPgpService2.java:62)
12-01 20:51:19.893 21610 21620 W System.err:    at org.openintents.openpgp.IOpenPgpService2$Stub.onTransact(IOpenPgpService2.java:80)
12-01 20:51:19.893 21610 21620 W System.err:    at android.os.Binder.execTransact(Binder.java:453)
12-01 20:51:19.893 21610 21620 E JavaBinder: Forcefully exiting
12-01 20:51:20.007   212   212 I Zygote  : Process 21610 exited cleanly (1)
12-01 20:51:20.014 21298 21644 E OpenPgp API: Exception in executeApi call
12-01 20:51:20.014 21298 21644 E OpenPgp API: android.os.DeadObjectException
12-01 20:51:20.014 21298 21644 E OpenPgp API:   at android.os.BinderProxy.transactNative(Native Method)
12-01 20:51:20.014 21298 21644 E OpenPgp API:   at android.os.BinderProxy.transact(Binder.java:503)
12-01 20:51:20.014 21298 21644 E OpenPgp API:   at org.openintents.openpgp.IOpenPgpService2$Stub$Proxy.execute(IOpenPgpService2.java:160)
12-01 20:51:20.014 21298 21644 E OpenPgp API:   at org.openintents.openpgp.util.OpenPgpApi.executeApi(OpenPgpApi.java:370)
12-01 20:51:20.014 21298 21644 E OpenPgp API:   at org.openintents.openpgp.util.OpenPgpApi.executeApi(OpenPgpApi.java:330)
12-01 20:51:20.014 21298 21644 E OpenPgp API:   at org.openintents.openpgp.util.OpenPgpApi$OpenPgpAsyncTask.doInBackground(OpenPgpApi.java:301)
12-01 20:51:20.014 21298 21644 E OpenPgp API:   at org.openintents.openpgp.util.OpenPgpApi$OpenPgpAsyncTask.doInBackground(OpenPgpApi.java:286)
12-01 20:51:20.014 21298 21644 E OpenPgp API:   at android.os.AsyncTask$2.call(AsyncTask.java:295)
12-01 20:51:20.014 21298 21644 E OpenPgp API:   at java.util.concurrent.FutureTask.run(FutureTask.java:237)
12-01 20:51:20.014 21298 21644 E OpenPgp API:   at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1113)
12-01 20:51:20.014 21298 21644 E OpenPgp API:   at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:588)
12-01 20:51:20.014 21298 21644 E OpenPgp API:   at java.lang.Thread.run(Thread.java:818)
12-01 20:51:20.017   780  1746 I ActivityManager: Process org.sufficientlysecure.keychain:remote_api_2 (pid 21610) has died
12-01 20:51:20.017   780  1746 D ActivityManager: cleanUpApplicationRecord -- 21610
12-01 20:51:20.017   780  1746 W ActivityManager: Scheduling restart of crashed service org.sufficientlysecure.keychain/.remote.OpenPgpService2 in 1000ms
```

**Comment by zeapo** [comment]
Interesting.
What version of OpenKeychain you use?

**Comment by cyrus-and** [comment]
- OpenKeychain: 4.1
- Password Store: 1.2.0.52
- Android: 6.0.1
- Nexus 5 (not rooted)

**Comment by dskrad** [comment]
I have can verify the same issue on Nougat. 

- OpenKeychain: 4.1
- Password Store: 1.2.0.52
- Android: 7.1.1
- Nexus 5X (not rooted)

I fixed this by downgrading to Password Store v1.2.0.51

**Comment by zeapo** [comment]
@cyrus-and can you confirm if you have this issue on 1.2.0.51?

**Comment by cyrus-and** [comment]
@zeapo Yes, same behavior as 1.2.0.52.

PS why does my settings are not remembered when I switch version? I also had to re-generate my SSH key.

**Comment by dskrad** [comment]
v1.2.0.51 seems not to have this issue. :)

I had to delete current version...install 1.2.0.51...then regenerate all settings and SSH keys and clone my password repo again. But I can now edit my stored passwords.

**Comment by zeapo** [comment]
Working on it right now:

@cyrus-and does "**Select OpenPGP Key id**" option in settings has a value?

**Comment by cyrus-and** [comment]
@zeapo It actually doesn't... :/
If I select the proper key then everything works (version 1.2.0.51).

**Comment by zeapo** [comment]
OK, that's what I thought. The .54 will contain only this option with a check that it's selected.

Thanks

**Comment by zeapo** [comment]
fixed in 5644c06f2a51bf495ceaab2a27b0482436093c31
Version .54 will contain this fix.
Thank you all for reporting.
